actioncable/lib/action_cable/connection/base.rb

```diff
@@ -91,8 +91,8 @@ def dispatch_websocket_message(websocket_message) #:nodoc:
         end
       end
 
-      def transmit(cable_message) # :nodoc:
-        websocket.transmit encode(cable_message)
+      def transmit(cable_message, encoded = false) # :nodoc:
+        websocket.transmit(encoded ? cable_message : encode(cable_message))
       end
```

Comment:
Boolean flags in method signature are a good red flag to investigate.

In this case, we really want a separate API to transmit already-encoded messages. The existing `#transmit` can invoke that API after encoding its message.